src/currentWeather/CurrentWeatherPanel.tsx

```diff
@@ -0,0 +1,50 @@
+import { useState } from "react";
+import { SearchPanel } from "./SearchPanel";
+import { CurrentWeatherAside } from "./CurrentWeatherAside";
+import "./CurrentWeatherPanel.css";
+import { CurrentWeather, ForecastWeather } from "../App";
+
+type CurrentWeatherPanelProps = {
+  searchHistory: string[];
+  fetchLocation: (location: string) => Promise<void>;
```

Comment:
We have talked about this more than 3 times I believe. 

CurrentWeatherPanel doesn't give a fuck, whether parent fetches something or not. It's incorrect from the perspective of data flow and architecture. 

Make prop `onSearch` and call it when SearchPanel calls `onSubmit`. Then you won't need promise type and so on.